cli/internal/cmd/recover.go

```diff
@@ -231,9 +191,9 @@ func (d *recoverDoer) setDialer(dialer grpcDialer, endpoint string) {
 	d.endpoint = endpoint
 }
 
-func (d *recoverDoer) setSecrets(getDiskKey func(string) ([]byte, error), measurementSecret []byte) {
-	d.getDiskKey = getDiskKey
-	d.measurementSecret = measurementSecret
+func (d *recoverDoer) setSecrets(masterSecret keyservice.MasterSecret) {
+	d.kmsURI = fmt.Sprintf(keyservice.ClusterKMSURI, masterSecret.EncodeKey(), masterSecret.EncodeSalt())
+	d.storageURI = keyservice.NoStoreURI
 }
```

Comment:
Not a fan of having a dedicated function just to parse the master secret into KMS URIs, especially considering that this will most likely be removed once the option for eKMS is implemented.
Maybe try to refactor this to be more general, or move the URI parsing logic into a switch/case block, so its easily extended for different KMS backends

Reply:
Are you fine with moving this to the PR that actually makes eKMS user facing? As you say it will change again very soon.

Reply:
Looked at the complete code of recover again:

The `setSecrets` method was used for two reasons:
1. We are working on an interface, so we should use setter functions to add values to it
2. We wanted an easy way to stub the dynamic key derivation function

With the switch from sending individual keys over a streaming rpc, to just sending the required information so the nodes can handle setup themselves, 2. is no longer necessary.
Instead the `setSecrets` method should be renamed to something like `setURIs` and take care of setting KMS and storage URI for the interface.

```Go
func (d *recoverDoer) setURIs(kmsURI, storageURI string) {
	d.kmsURI = kmsURI
	d.storageURI = storageURI
}
```